src/main/java/useCases/createListing/CreateListingInteractor.java

```diff
@@ -0,0 +1,85 @@
+package useCases.createListing;
+
+import entities.Listing;
+import entities.ListingFactory;
+import entities.LoggedInUserSingleton;
+import intefaceAdapters.createListing.ListingRepo;
+
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.util.UUID;
+
+public class CreateListingInteractor implements CreateListingInputBoundary{
+    private CreateListingDsGateway gateway;
+    private CreateListingPresenter presenter;
+    private ListingFactory listingFactory;
+
+    public CreateListingInteractor(CreateListingPresenter presenter, ListingFactory listingFactory) {
+        this.presenter = presenter;
+        try {
+            gateway = new ListingRepo("./listings.csv");
+        } catch (IOException e) {
+            throw new RuntimeException("Could not find file");
+        }
+        this.listingFactory = listingFactory;
+    }
+
+    @Override
+    public void create(CreateListingRequestModel requestModel) {
+        String uniqueId = UUID.randomUUID().toString();
```

Comment:
I see you've validated a few fields well but we should definitely make sure to not allow the user to leave the rest of the fields empty. Right now a user can just create a listing with no name or no brand.